web/src/user_group_edit.ts

```diff
@@ -1533,28 +1557,94 @@ export function maybe_reset_right_panel(groups_list_data: UserGroup[]): void {
     // Check if we have any groups in panel to decide whether to
     // display a notice.
-    let has_groups;
     const is_your_groups_tab_active =
         get_active_data().$tabs.first().attr("data-tab-key") === "your-groups";
+
+    let has_groups;
+    const current_group_filter =
+        filters_dropdown_widget.value() ?? FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
+    const include_deactivated_groups =
+        current_group_filter === FILTERS.DEACTIVATED_GROUPS ||
+        current_group_filter === FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
     if (is_your_groups_tab_active) {
-        has_groups = user_groups.get_user_groups_of_user(people.my_current_user_id(), true).length;
+        const groups = user_groups.get_user_groups_of_user(
+            people.my_current_user_id(),
+            include_deactivated_groups,
+        );
+        if (current_group_filter === FILTERS.DEACTIVATED_GROUPS) {
+            has_groups = user_subscribed_to_deactivated_user_groups();
+        } else {
+            has_groups = groups.length;
+        }
     } else {
-        has_groups = user_groups.get_realm_user_groups().length;
+        const realm_user_groups = user_groups.get_realm_user_groups(include_deactivated_groups);
+
+        if (current_group_filter === FILTERS.DEACTIVATED_GROUPS) {
+            has_groups = realm_has_deactivated_user_groups();
+        } else {
+            has_groups = realm_user_groups.length;
+        }
     }
     if (has_groups) {
         $(".no-groups-to-show").hide();
         return;
     }
-    if (is_your_groups_tab_active) {
-        $(".all_groups_tab_empty_text").hide();
-        $(".your_groups_tab_empty_text").show();
-    } else {
-        $(".your_groups_tab_empty_text").hide();
-        $(".all_groups_tab_empty_text").show();
+
+    let empty_user_group_list_message: string;
+    switch (current_group_filter) {
+        case FILTERS.DEACTIVATED_GROUPS:
+            if (is_your_groups_tab_active) {
+                empty_user_group_list_message =
+                    EMPTY_USER_GROUP_LIST_NOTICES.YOUR_GROUPS_DEACTIVATED_NONE;
+            } else {
+                empty_user_group_list_message =
+                    EMPTY_USER_GROUP_LIST_NOTICES.ALL_GROUPS_DEACTIVATED_NONE;
+            }
+            break;
+        case FILTERS.ACTIVE_GROUPS:
+            if (is_your_groups_tab_active) {
+                empty_user_group_list_message =
+                    EMPTY_USER_GROUP_LIST_NOTICES.YOUR_GROUPS_ACTIVE_NONE;
+            } else {
+                empty_user_group_list_message =
+                    EMPTY_USER_GROUP_LIST_NOTICES.ALL_GROUPS_ACTIVE_NONE;
+            }
+            break;
+        case FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS:
+            if (is_your_groups_tab_active) {
+                empty_user_group_list_message = EMPTY_USER_GROUP_LIST_NOTICES.YOUR_GROUPS_NONE;
+            } else {
+                empty_user_group_list_message = EMPTY_USER_GROUP_LIST_NOTICES.ALL_GROUPS_NONE;
```

Comment:
There is a bug where no message is shown while typing in the search input. On that note I was thinking if we could just have the message like "There are no groups matching your filter" for the following cases -
- If search input is non empty.
- Drodpown filter is shown and filter is not set to "Active and deactivated groups".